exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,330 @@
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Options
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key. Note: the specific handling depends on the option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Duplicate Options Management
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key and value. Note: the specific handling depends on the option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Script Option Removal
+`req~script-option-removal~1`
+
+The parser handler must remove found known Script Options from the original script code. Successfully parsed Script Options, which are not recognized by the parser handler shall remain in the script code.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+- `feat~java-specific-script-options~1`
+
+
+### Escape Sequence Script Options Parsing
+`req~escape-sequence-script-options-parsing~1`
+
+The following rules for escape sequences at any place within a script optionValue are to be applied:
+- '\n' => <line feed> character
+- '\r' => <carriage return> character
+- '\;' => ';' character
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+
+### Java %scriptclass Option Handling V1
+`req~java-scriptclass-option-handling-v1~1`
+
+The Java parser handler must correctly identify the first `%scriptclass` option and remove only this single instance from the script code. Any further occurrences of `%scriptclass` option shall stay in the source script code.
+The value should be handled according to the [Java specification for identifies](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8).
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+
+### Java %scriptclass Option Handling V2
+`req~java-scriptclass-option-handling-v2~1`
+
+The Java parser handler must correctly identify the first `%scriptclass` option and remove any additional occurrences of this option within the script code.
+The value should be handled according to the [Java specification for identifies](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8).
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling V1
+`req~java-jar-option-handling-v1~1`
+
+The Java parser handler must find multiple %jar options. The values are to be interpreted as the Java CLASSPATH: `<file1>:<file2>:...:<filen>`.
+The Java parser handler shall split the entries by the colon character. The Java parser handler shall identify duplicated files and order the result of all `%jar` options alphabetically.
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling V2
+`req~java-jar-option-handling-v2~1`
+
```

Comment:
I have the feeling both %jar option handling should be split and the first part can be common requirements for both

Reply:
Done